velox/functions/prestosql/aggregates/tests/AverageAggregationTest.cpp

```diff
@@ -199,5 +199,120 @@ TEST_F(AverageAggregationTest, partialResults) {
 
   assertQuery(plan, "SELECT row(4950, 100)");
 }
+
+TEST_F(AverageAggregationTest, partialAvgForNulls) {
+  auto batchSize = 1'000;
+  auto rowType = ROW({"c0", "c1", "c2"}, {INTEGER(), INTEGER(), DOUBLE()});
+
+  auto vec = makeFlatVector<double>(batchSize, [](auto row) { return row; });
+  for (auto i = 0; i < vec->size() - 1; i++) {
+    vec->setNull(i, true);
+  }
+
+  auto data = makeRowVector(
+      {makeFlatVector<int32_t>(
+           batchSize, [](auto row) { return row % 100; }, nullEvery(6)),
+       makeFlatVector<int32_t>(
+           batchSize, [](auto row) { return row % 5; }, nullEvery(8)),
+       vec});
+  createDuckDbTable({data});
+
+  auto planNode =
+      PlanBuilder()
+          .values({data})
+          .project({"c0", "c2"})
+          .addNode([&](std::string id, core::PlanNodePtr input) {
```

Comment:
It is not clear to me what is the scenario that doesn't work. Would you explain? Why can't you use PlanBuilder's aggregation methods here? What kind of node are you trying to add?

Reply:
Sorry for the misunderstanding. I just want to create a partial avg and select the sum and count from the row vector. But found if creating with plan builder,  the row vector had empty names for its chidren columns, and I do not know how to select the child column from it. So I just used the addNode way to assign names for the sum and count columns for later selection.
The case I want to create is like below:
```
.partialAggregation({"c0"}, {"avg(p0) as a0"})
.project({"c0", "a0.?", "a0.?"})
```
This test can be removed if there is no need to align the result with duckdb in this case.